examples/src/main/java/com/mapbox/navigation/examples/core/MapboxBuildingHighlightActivity.kt

```diff
@@ -83,20 +76,34 @@ class MapboxBuildingHighlightActivity : AppCompatActivity(), OnMapLongClickListe
     private val waypointsController = WaypointsController()
 
     /**
-     * There are two available apis for highlighting buildings. The MapboxBuildingHighlightApi
-     * allows you to select buildings. When running this example, try tapping on a building
-     * to see it highlight.
-     *
-     * The MapboxBuildingArrivalApi will automatically highlight a building when the navigation
-     * has arrived.
+     * This api allows you to query a building feature if it exists on a [MapboxMap]
      */
-    private val buildingsArrivalApi = MapboxBuildingArrivalApi()
-    private lateinit var buildingHighlightApi: MapboxBuildingHighlightApi
+    private val buildingsApi: MapboxBuildingsApi by lazy {
+        MapboxBuildingsApi(mapboxMap)
+    }
+
+    /**
+     * This view can be used to render building highlight if it obtained as a result from the query
+     * using [MapboxBuildingsApi]
+     */
+    private val buildingView = MapboxBuildingView()
 
     /**
-     * Access the current state of the extruded buildings layer.
+     * The callback contains a list of buildings returned as a result of querying the [MapboxMap].
+     * If no buildings are available, the list is empty
      */
-    private var extrudeBuildings: Boolean = false
+    private val callback =
+        MapboxNavigationConsumer<Expected<BuildingError, BuildingValue>> { expected ->
+            expected.fold(
```

Comment:
nit
```suggestion
            expected.onValue(
```

Reply:
Or it could be good to add a log and comment in the error block. To show where to capture "no building highlighted" events